static/js/lib/validation.js

```diff
@@ -32,16 +32,43 @@ export const resetPasswordFormValidation = yup.object().shape({
 })
 
 export const changePasswordFormValidation = yup.object().shape({
+  email: emailFieldValidation.label("Email"),
+
+  user_email: yup.string().label("User Email"),
+
+  /* eslint-disable camelcase */
+  emailPassword: yup
+    .string()
+    .label("Confirm Password")
+    .when(["email", "user_email"], (email, user_email, schema) =>
+      user_email !== email ? schema.required().min(8) : schema.notRequired()
+    ),
+
   oldPassword: yup
     .string()
     .label("Old Password")
-    .required(),
+    .when(["email", "user_email"], (email, user_email, schema) =>
+      user_email === email ? schema.required() : schema.notRequired()
+    ),
 
-  newPassword: newPasswordFieldValidation.label("New Password"),
+  newPassword: newPasswordFieldValidation
+    .label("New Password")
+    .when(["email", "user_email"], (email, user_email, schema) =>
+      user_email === email
+        ? schema
+          .matches(/^(?=.*[0-9])(?=.*[a-zA-Z]).*$/, {
+            message: "Password must contain at least one letter and number"
+          })
+          .required()
+          .min(8)
+        : schema.matches(/^.*$/).notRequired()
```

Comment:
I don't think these extra validations on password were in scope for this PR

Reply:
We need that validation because we need to make it optional if the user has the intention to only change the email address.